test/e2e/cert_manager_deployment_test.go

```diff
@@ -375,7 +375,7 @@ func verifyValidControllerOperatorStatus(t *testing.T, client *certmanoperatorcl
 
 func addValidControlleDeploymentConfig(operator *v1alpha1.CertManager) {
 	operator.Spec.ControllerConfig = &v1alpha1.DeploymentConfig{
-		OverrideArgs: []string{"--dns01-recursive-nameservers=10.10.10.10:53", "--dns01-recursive-nameservers-only"},
+		OverrideArgs: []string{"--dns01-recursive-nameservers=10.10.10.10:53", "--dns01-recursive-nameservers-only", "--enable-certificate-owner-ref"},
```

Comment:
I think we should refrain from adding this flag here, as it would mean that every e2e test would end up using certificate-owner-ref which isn't something we want to test (IMO better not to change default behaviour for now) rather we add e2e(s) specific to this scenario as a follow-up PR to specifically test the effects of applying this flag (similar to discussion we had today).

`"--dns01-recursive-nameservers=10.10.10.10:53", "--dns01-recursive-nameservers-only"` can still remain IMHO because some certificate tests on AWS explicitly needs these flags.

Reply:
The function `addValidControlleDeploymentConfig` is only used once inside `TestContainerOverrides` e2e test and not all e2e tests. Also these flags are added once and the validity of the status of the cert-manager operator is checked. After that the flags are removed. Thus this flags do not effect any other tests. 

When we modularize the e2e test, we anyway need to add the flag `--enable-certificate-owner-ref` to check whether it is correctly working or not. Adding it here, IMO, verifies that it is working correctly and makes the PR complete.

Let me know your thoughts.

Reply:
Cool, then we're good to keep it here for now.